### ISSUE TITLE
[Question]: Does llama-index settings even work?

### ISSUE THREAD

**Issue by Aekansh-Ak** [run-llama/llama_index#12702]
### Question Validation

- [X] I have searched both the documentation and discord for an answer.

### Question

My code-:

#########################################################
reader = SimpleDirectoryReader(
    input_files=["/home/seceon/opensearch_lm_index/textdocs/aktripathi.txt"]
)
docs = reader.load_data()
text = docs[0].text

summarizer = TreeSummarize()
embeddings = HuggingFaceEmbedding()

text_split = RecursiveCharacterTextSplitter(chunk_size=300, chunk_overlap=190)
text_splitter = LangchainNodeParser(text_split)
Settings.text_splitter = text_splitter
nodes = text_splitter(documents)

llm = HuggingFaceLLM(
    model_name=model_id,
    tokenizer_name=model_id,
    context_window=3900,
    max_new_tokens=256,
    model_kwargs={"torch_dtype": torch.float16},
    # tokenizer_kwargs={},
    generate_kwargs={"temperature": 0.3,  "top_p": 0.95},
    device_map="auto",
)

Settings.llm = llm
Settings.embed_model = embeddings
Settings.context_window=4096,
Settings.num_output=1000,
Settings.chunk_overlap_ratio=0.3,
Settings.chunk_size_limit=None

response = summarizer.aget_response("Summarise the data in the document", [text])
print(response)
####################################################################

Error-:

ValueError: 
******
Could not load OpenAI model. If you intended to use OpenAI, please check your OPENAI_API_KEY.
Original error:
No API key found for OpenAI.
Please set either the OPENAI_API_KEY environment variable or openai.api_key prior to initialization.
API keys can be found or created at https://platform.openai.com/account/api-keys

To disable the LLM entirely, set llm=None.

Does the settings.llm even work?

Reference-:
https://docs.llamaindex.ai/en/stable/examples/response_synthesizers/tree_summarize/

**Comment by logan-markewich** [comment]
@Aekansh-Ak You created the summarizer before changing the settings. Try swapping the order

More technically
- creating the summarize will pull from the current global settings to create an LLM object
- at this point in change, the settings haven't changed, so it pulls openai
- then you modify the settings
- however, the summarizer is already created with an LLM object in memory attached to it, it is not updated

**Comment by Aekansh-Ak** [comment]
Thanks! That worked.

For this line-:

response = await summarizer.aget_response("who is Paul Graham?", [text])

I am getting-:

 response = await summarizer.aget_response("who is Paul Graham?", [text], llm=None)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
SyntaxError: 'await' outside function

**Comment by Aekansh-Ak** [comment]
Thanks Dosu bot, the llama-index documentation clearly can't be just blindly followed.

**Comment by logan-markewich** [comment]
@Aekansh-Ak In python, `await` can only be used if you are running in an async function. Our examples in the docs mostly come from python jupyter notebooks, which are already running in async by default, and therefore can use `await`.

**Comment by Aekansh-Ak** [comment]
Okay, understood.

How can I do streaming in this, it doesn't work-:

summarizer = TreeSummarize()
async def main():
    response = await summarizer.aget_response("Could you summarize the given context? Return your response which covers the key points of the text.", [text], streaming = True)
    for r in response:
      print(str(r), end="", flush=True)
      
AttributeError: 'str' object has no attribute 'response_gen'

**Comment by logan-markewich** [comment]
The streaming kwarg goes in the constructor
`TreeSummarize(streaming=True)`
